### ISSUE TITLE
Clarify denomination for different system types

### ISSUE THREAD

**Issue by sitangshugk95** [ContactEngineering/Adhesion#76]
It would be helpful to clarify what each system means and when to use what: 

- NonSmoothContactSystem vs SmoothContactSystem

- BoundedSmoothContactSystem vs SmoothContactSystem

**Comment by sitangshugk95** [comment]
@sannant Having some brief explanations about these denominations would be helpful (whenever convenient for you, or whoever is actively working with this package). Thanks again.

**Comment by sannant** [comment]
The denomination `HardWall` vs `SoftWall` in the potentials 

### NonSmoothContactSystem -> HardWallContactSystem
   It is a contact system with hardwall repulsion and no interactions.
      
###  SmoothContactSystem -> SoftWallCohesiveContactSystem 
     A contacsystem that has an attractive cohesive law and a soft repulsion 
     This is for interactions like the Lennard-Jones potential  

###  BoundedSmoothContactSystem -> HardWallCohesiveContactSystem
   A contacsystem that has a cohesive law and a hardwall repulsion 
   It can in principle also additionally have SoftWall interaction. 

   I.e. One can combine the Lennard-Jones interaction with the HardWall constraint.


## Comments 

I hesitate to call "SoftWallCohesiveContactSystem" , "SoftWallContactSystem".

However, we could reserve SoftWallContactSystem for system with soft repulsion with no adhesion, which are guaranteed to bw convex, and can be safely used with simple minimizers. 

@pastewka , do you have better ideas ? 
@sitangshugk95 , is it clearer?

**Comment by sitangshugk95** [comment]
@sannant This is very clear, yes. Wondering if this can be moved to "Discussion" as it can be of help to future users.

**Comment by sannant** [comment]
Well, we should actually implement this renaming and write this in the documentation. I want to keep it as an issue as a "TODO" item

**Comment by sitangshugk95** [comment]
> The denomination `HardWall` vs `SoftWall` in the potentials
> 
> ### NonSmoothContactSystem -> HardWallContactSystem
> It is a contact system with hardwall repulsion and no interactions.
> 
> ### SmoothContactSystem -> SoftWallCohesiveContactSystem
> ```
>  A contacsystem that has an attractive cohesive law and a soft repulsion 
>  This is for interactions like the Lennard-Jones potential  
> ```
> 
> ### BoundedSmoothContactSystem -> HardWallCohesiveContactSystem
> A contacsystem that has a cohesive law and a hardwall repulsion It can in principle also additionally have SoftWall interaction.
> 
> I.e. One can combine the Lennard-Jones interaction with the HardWall constraint.
> Would one typically do it because it is easier to converge?
> ## Comments
> I hesitate to call "SoftWallCohesiveContactSystem" , "SoftWallContactSystem".
> 
> However, we could reserve SoftWallContactSystem for system with soft repulsion with no adhesion, which are guaranteed to bw convex, and can be safely used with simple minimizers.
> 
> @pastewka , do you have better ideas ? @sitangshugk95 , is it clearer?

**Comment by sannant** [comment]
Yes. Usually one wants stiff repulsion. These are hard to deal with because they lead to bad condition number of the Hessian matrix. It is better to approximate the stiff repulsion by a hardwall constraint dealt with in the minimizer. The constrained minimizers are a bit more complex but it is worth it.

**Comment by pastewka** [comment]
I agree, we need to think about the naming conventions in the code and also make the whole system construction easier to use and more transparent.
